cloud-chatops/src/features/base_feature.py

```diff
@@ -0,0 +1,187 @@
+# then we set the author as a default value.
+DEFAULT_AUTHOR = "U01JG0LKU3W"  # David Fairbrother
+DEFAULT_CHANNEL = "C06U37Y02R4"  # "dev-chatops" channel as default
+DEFAULT_REPO_OWNER = "stfc"  # Our repos are owned by "stfc"
+
+
+class BaseFeature:
+    """This base class provides universal methods for features."""
+
+    # pylint: disable=R0903
+    # This is a base feature and not intended to be used on its own
+    channel: str
+
+    def __init__(self):
+        self.channel = DEFAULT_CHANNEL
+        self.client = WebClient(token=get_token("SLACK_BOT_TOKEN"))
+        self.prs = GetGitHubPRs(get_repos(), DEFAULT_REPO_OWNER).run()
+        self.slack_ids = get_user_map()
+
+    def _github_to_slack_username(self, user: str) -> str:
+        """Makes a call to the global method"""
+        return _github_to_slack_username(self.slack_ids, user)
+
+    def _slack_to_human_username(self, slack_member_id: str) -> str:
+        """Makes a call to the global method"""
+        return _slack_to_human_username(self.client, slack_member_id)
+
+    def _post_reminder_message(self) -> str:
+        """
+        This method posts the main reminder message to start the thread of PR notifications.
+        :return: The reminder message object
+        """
+        reminder = self.client.chat_postMessage(
+            channel=self.channel,
+            text="Here are the outstanding PRs as of today:",
+        )
+        assert reminder["ok"]
+        return reminder.data["ts"]
+
+    def _send_no_prs_found(self, thread_ts) -> None:
+        """
+        This method sends a message to the chat that there are no PRs.
+        :param thread_ts: Timestamp of reminder message
+        """
+        response = self.client.chat_postMessage(
+            text="There are no outstanding pull requests open.",
+            channel=self.channel,
+            thread_ts=thread_ts,
+            unfurl_links=False,
+        )
+        assert response["ok"]
+
+    def _send_thread(
+        self, pr_data: PrData, thread_ts: str
+    ) -> WebClient.chat_postMessage:
+        """
+        This method sends the message and returns the response.
+        :param pr_data: The PR data as a dataclass
+        :param thread_ts: Timestamp of reminder message
+        :return: The message response.
+        """
+        message = PRMessageBuilder().make_message(pr_data)
+        response = self.client.chat_postMessage(
+            text=message,
+            channel=self.channel,
+            thread_ts=thread_ts,
+            unfurl_links=False,
+        )
+        assert response["ok"]
+        return response
+
+    def _send_thread_react(self, pr: PrData, channel: str, thread_ts: str) -> None:
+        """
+        This method sends reactions to the PR message if necessary.
+        """
+        react_with = []
+        if pr.old:
+            react_with.append("alarm_clock")
+        if pr.draft:
+            react_with.append("scroll")
```

Comment:
Some bike shedding - I thought `scroll` was comments pending not draft
Could we do something like `building_construction` (a crane) as it's clearer to me at least it's not finished?

Reply:
Good idea